src/maxdiffusion/models/train.py

```diff
@@ -526,10 +528,12 @@ def compute_loss(unet_params):
         if step == last_profiling_step:
             max_utils.deactivate_profiler(config)
 
-        mllog_utils.maybe_train_step_log(config, start_step, step_num, samples_count, train_metric)
-
-    steptime = (time.time() - start_time)/ (config.max_train_steps - start_step) * 1000
-    max_logging.log(f"avg step time of {max_train_steps}, {steptime} ms")
+    if config.write_metrics:
+        # log the last metrics_period
```

Comment:
we probably also need to update `record_scalar_metrics` here?

Reply:
write metrics is always one period behind record_scalar_metrics. In the training loop, we call both record_scalar_metrics and write_metrics to have current period metrics recorded, but print the previous period metrics. when the training loop finishes, all the periods have been recorded, so we only need to print the late period metrics. I updated and applied the same logic to mllog_utils.maybe_train_step_log